junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/execution/JupiterEngineExecutionContext.java

```diff
@@ -37,9 +38,9 @@ public class JupiterEngineExecutionContext implements EngineExecutionContext {
 	private boolean beforeAllCallbacksExecuted = false;
 	private boolean beforeAllMethodsExecuted = false;
 
-	public JupiterEngineExecutionContext(EngineExecutionListener executionListener,
-			JupiterConfiguration configuration) {
-		this(new State(executionListener, configuration));
+	public JupiterEngineExecutionContext(EngineExecutionListener executionListener, JupiterConfiguration configuration,
+			OutputDirProvider outputDirProvider) {
```

Comment:
You know this design much better than I do, so this is just a question: would it be better to make OutputDirProvider an attribute of JupiterConfiguration rather than passing it alongside?

Reply:
It was passed alongside because `JupiterConfiguration` is created during discovery while `OutputDirProvider` was only passed in later as part of `ExecutionRequest`. However, since `Launcher` has an `execute` method that takes a discovery request, I've moved it there in 170bfa606f63d103e561a94f537e01295a74e668 which did simplify the code considerably.